location/OBARegionHelper.m

```diff
@@ -90,45 +91,47 @@ - (void)setNearestRegion {
 
         [self.regions removeObjectsInArray:regionsToRemove];
 
-        if (self.regions.count == 0) {
-            [self.appDelegate.modelDao writeSetRegionAutomatically:NO];
-            [self.appDelegate.locationManager removeDelegate:self];
-            [self.appDelegate showRegionListViewController];
-            return;
+        if (self.regions.count != 0) {
+            [self.regions
+             sortUsingComparator:^(id obj1, id obj2) {
```

Comment:
can be simplified into `sortUsingComparator:^(OBARegionV2 *region1, OBARegionV2 *region2)`